### PR TITLE
Function pointer tests

### PR DESCRIPTION
**Pull request by cston**
See #5412

### CODE REVIEW

**Comment by cston** [comment]
@ericfe-ms, @amcasey, @kevinh-ms, @pnelsonmsft

File: src/ExpressionEvaluator/CSharp/Test/ResultProvider/FunctionPointerTests.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+using Microsoft.VisualStudio.Debugger.Evaluation;
+using System.Diagnostics;
+using Xunit;
+using Type = Microsoft.VisualStudio.Debugger.Metadata.Type;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+{
+    public class FunctionPointerTests : CSharpResultProviderTestBase
+    {
+        [Fact(Skip = "https://github.com/dotnet/roslyn/issues/5412")]
```
Comment (amcasey) [inline src/ExpressionEvaluator/CSharp/Test/ResultProvider/FunctionPointerTests.cs]: Also add a WorkItem attribute?

File: src/ExpressionEvaluator/Core/Source/ResultProvider/Formatter.Values.cs [hunk 2]
```diff
@@ -373,7 +373,6 @@ internal string FormatPrimitive(DkmClrValue value, ObjectDisplayOptions options,
             }
             else
             {
-                Debug.Assert(value.HostObjectValue != null);
```
Comment (amcasey) [inline src/ExpressionEvaluator/Core/Source/ResultProvider/Formatter.Values.cs]: Could we keep the assert and add "or is function pointer"?

File: src/ExpressionEvaluator/Core/Test/ResultProvider/Debugger/Engine/DkmClrValue.cs [hunk 3]
```diff
@@ -39,8 +39,6 @@ public class DkmClrValue : DkmDataContainer
         {
             Debug.Assert(!type.GetLmrType().IsTypeVariables() || (valueFlags == DkmClrValueFlags.Synthetic));
             Debug.Assert((alias == null) || evalFlags.Includes(DkmEvaluationResultFlags.HasObjectId));
-            // The "real" DkmClrValue will always have a value of zero for null pointers.
-            Debug.Assert(!type.GetLmrType().IsPointer || (value != null));
```
Comment (amcasey) [inline src/ExpressionEvaluator/Core/Test/ResultProvider/Debugger/Engine/DkmClrValue.cs]: Could we update the assert?

**Comment by amcasey** [comment]
:+1:

**Comment by KevinH-MS** [comment]
:+1:

File: src/ExpressionEvaluator/Core/Source/ResultProvider/Formatter.Values.cs (hunk 2 above)
Comment (cston) [inline src/ExpressionEvaluator/Core/Source/ResultProvider/Formatter.Values.cs]: I'll revert this change for now. The assert can be updated when IsFunctionPointer is available. Same with the other assert below.

**Comment by cston** [comment]
@dotnet-bot retest this please

**Pull request merged**